src/inmanta/ast/statements/__init__.py

```diff
@@ -149,15 +234,103 @@ def get_dataflow_node(self, graph: DataflowGraph) -> dataflow.NodeReference:
 
 
 class Resumer(ExpressionStatement):
+    """
+    Resume on a set of requirement variables' values when they become ready (i.e. they are complete).
+    """
+
     def resume(self, requires: Dict[object, object], resolver: Resolver, queue: QueueScheduler, target: ResultVariable) -> None:
         pass
 
 
 class RawResumer(ExpressionStatement):
+    """
+    Resume on a set of requirement variables when they become ready (i.e. they are complete).
+    """
+
     def resume(self, requires: Dict[object, ResultVariable], resolver: Resolver, queue: QueueScheduler) -> None:
         pass
 
 
+class VariableReferenceHook(RawResumer):
```

Comment:
This rename is step forward. If you feel like you could name these things better than 'Resumer' and 'Hangunit', feel free to rename, as I don't consider these names to be very good.